### ISSUE TITLE
Different Pickers With Same Relation Type Don't Sync?

### ISSUE THREAD

**Issue by s6admin** [uComponents/nuPickers#62]
We're seeing a major problem with syncing Pickers across Document Types.

We are trying to relate different document types (such as a Product and a Download) using a bidirectional "relation only" data type. We actually have two different pickers, one for each end of the relationship, but have them both mapped to the same Relation Type. The picker for the Product node has a list of Downloads that can be related. The picker for the Download node has a list of Products that can be related. These two lists of Products and Downloads are the same parent/child ids that get saved in the Relation table, but the two pickers do not sync. We can only see the checkboxes from the childId node where the selections were originally made.

If we only use one nuPicker then the values are synced, but that isn't practical because we need a different set of checkboxes depending on which end of the relationship the admin is making the selections (product node or download node).

Has anyone else encountered this? Is this a limitation of nuPickers or are we implementing it wrong?
Thank you!

**Comment by amainguy** [comment]
Removing this line will make it work: 

/nuPickers/Shared/RelationMapping/RelationMappingApiController.cs -- line 46.

it checks for the data type definition Id. (in other words, you have to use the same datatypes if you wan't them to sync). I would make this line optionnal (within the configuration section) If I were the owner of the project.

**Comment by Hendy** [comment]
Hi, sorry for the late reply. What would you call such a config option, _Lock to Data Type_ ? I image it'd be a check box that appears when "relations only" is selected, and it's probably better to default to not executing [that line](https://github.com/uComponents/nuPickers/blob/master/source/nuPickers/Shared/RelationMapping/RelationMappingApiController.cs#L46)

**Comment by enkelmedia** [comment]
This should also be included in upcoming release =D

**Comment by s6admin** [comment]
Agreed!

**Comment by hfloyd** [comment]
+1
I have the same issue - two different doctypes which are related, and I'd like to use synchronized prickers between them, so that an editor can select the relationship from either "side" - whichever is more convenient.

I'll try compiling my own version of nuPickers with that line commented-out... but I do hope it makes it to the next "official" release.

(Ps. Using Umbraco v7.3.4)

**Comment by hfloyd** [comment]
Ah, found it. 
File:
\source\nuPickers\Shared\RelationMapping\RelationMapping.cs

Line 37:
relations = relations.Where(x => new RelationMappingComment(x.Comment).DataTypeDefinitionId == relationMappingComment.DataTypeDefinitionId);

Commenting that one line and recompiling fixes the issue for me :-)

**Comment by SayTen** [comment]
+1
I'd be happy to do a pull request for this if understood the intention of the comment filtering.

**Comment by s6admin** [comment]
Yes, each time we begin a new project this issue gets in the way. I'm not sure the value of mapping to a bidirectional relationship if we can't let content editors benefit from it. Asking them to "remember" which node they can make the relationship from is terrible for usability. Anyone able to at least provide some general idea if this item will be addressed?
